livingpark_utils/download/ppmi.py

```diff
@@ -0,0 +1,270 @@
+"""Downloader for the ppmi dataset."""
+import os.path
+
+import pandas as pd
+import ppmi_downloader
+
+from .DownloaderABC import DownloaderABC
+from livingpark_utils.dataset import ppmi
+
+
+class Downloader(DownloaderABC):
+    """Handle the download of PPMI dataset.
+
+    Parameters
+    ----------
+    DownloaderABC : DownloaderABC
+        Abstract class to handle dataset download.
+    """
+
+    def __init__(
+        self, out_dir: str, *, cache_dir: str = ".cache", headless=True
+    ) -> None:
+        """Initialize a download handler.
+
+        During initialization, the output and cache directories are created.
+
+        Parameters
+        ----------
+        out_dir : str
+            Path of the output directory.
+        cache_dir : str, optional
+            Path of the cache directory., by default ".cache"
+        """
+        super().__init__(out_dir, cache_dir)
+        self.headless = headless
+
+    def get_metadata(
+        self,
+        query: list[str],
+        *,
+        timeout: int = 600,
+    ) -> tuple[list[str], list[str]]:
+        """Download PPMI required study files, if not available.
+
+        Parameters
+        ----------
+        query : list
+            Required PPMI study files (cvs files) supported by ppmi_downloader.
+        timeout : int, default 600
+            Number of second before the download times out.
+
+        Raises
+        ------
+        Exception:
+            If failure occurs during download.
+        """
+        try:
+            downloader = ppmi_downloader.PPMIDownloader()
+            downloader.download_metadata(
+                query,
+                destination_dir=self.out_dir,
+                headless=self.headless,
+                timeout=timeout,
+            )
+        except Exception:
+            missing = self.missing_metadata(query)
+            success = list(set(query) - set(missing))
+            return success, missing
+
+        return query, []
+
+    def missing_metadata(self, query, *, force: bool = False) -> list[str]:
```

Comment:
metadata should be renamed to `study_data`

Reply:
Fixed in new commit